sys/include/ztimer64.h

```diff
@@ -0,0 +1,519 @@
+    ztimer_clock_t *base_clock;     /**< 32bit clock backend                */
+    ztimer_t base_timer;            /**< 32bit backend timer                */
+    uint32_t base_last;             /**< lower timer last checkpoint time   */
+    uint64_t checkpoint;            /**< lower timer checkpoint offset      */
+    uint16_t adjust_set;            /**< will be subtracted on every set()  */
+    uint16_t adjust_sleep;          /**< will be subtracted on every sleep(),
+                                         in addition to adjust_set          */
+#if MODULE_PM_LAYERED || DOXYGEN
+    uint8_t block_pm_mode;          /**< min. pm mode to block for the clock to run */
+#endif
+};
+
+/**
+ * @brief   Get the current time from a clock
+ *
+ * @param[in]   clock          ztimer clock to operate on
+ *
+ * @return  Current count on @p clock
+ */
+uint64_t ztimer64_now(ztimer64_clock_t *clock);
+
+/**
+ * @brief   Get absolute target time for a clock given offset
+ *
+ * @param[in]   clock       ztimer clock to operate on
+ * @param[in]   offset      relative target time
+ *
+ * @returns absolute target time
+ */
+static inline uint64_t ztimer64_offset2absolute(ztimer64_clock_t *clock,
+                                                uint64_t offset)
+{
+    unsigned state = irq_disable();
+    uint64_t result = ztimer64_now(clock) + offset;
+
+    irq_restore(state);
+    return result;
+}
+
+/**
+ * @brief   Set a timer on a clock (absolute version)
+ *
+ * This will place @p timer in the timer targets queue of @p clock.
+ *
+ * @note The memory pointed to by @p timer is not copied and must
+ *       remain in scope until the callback is fired or the timer
+ *       is removed via @ref ztimer64_remove
+ *
+ * @param[in]   clock       ztimer clock to operate on
+ * @param[in]   timer       timer entry to set
+ * @param[in]   target      absolute target time
+ */
+void ztimer64_set_at(ztimer64_clock_t *clock, ztimer64_t *timer,
+                     uint64_t target);
+
+/**
+ * @brief   Set a timer on a clock (relative version)
+ *
+ * This will place @p timer in the timer targets queue of @p clock.
+ *
+ * @note The memory pointed to by @p timer is not copied and must
+ *       remain in scope until the callback is fired or the timer
+ *       is removed via @ref ztimer64_remove
+ *
+ * @param[in]   clock       ztimer clock to operate on
+ * @param[in]   timer       timer entry to set
+ * @param[in]   offset      relative target time
+ */
+static inline void ztimer64_set(ztimer64_clock_t *clock, ztimer64_t *timer,
+                                uint64_t offset)
+{
+    ztimer64_set_at(clock, timer,
+                    ztimer64_offset2absolute(clock, offset));
+}
+
+/**
+ * @brief   Check if a timer is currently active
+ *
+ * @param[in]   timer       timer to check
+ *
+ * @return 1 if timer is active
+ * @return 0 if timer is not active
+ */
+unsigned ztimer64_is_set(const ztimer64_t *timer);
```

Comment:
```suggestion
 * @return true if timer is active
 * @return false if timer is not active
 */
bool ztimer64_is_set(const ztimer64_t *timer);
```

Reply:
+1 in general, but `ztimer_is_set()` also returns unsigned 1/0.